skyportal/tests/api/test_sources.py

```diff
@@ -824,6 +824,167 @@ def test_source_photometry_summary_info(
+            'diameter': 10.0,
+        },
+        token=super_admin_token,
+    )
+    assert status == 200
+    assert data['status'] == 'success'
+    telescope_id = data['data']['id']
+
+    instrument_name = str(uuid.uuid4())
+    status, data = api(
+        'POST',
+        'instrument',
+        data={
+            'name': instrument_name,
+            'type': 'imager',
+            'band': 'NIR',
+            'filters': ['ztfg'],
+            'telescope_id': telescope_id,
+        },
+        token=super_admin_token,
+    )
+    assert status == 200
+    assert data['status'] == 'success'
+    instrument_id = data['data']['id']
+
+    # Some very high mjd to make this the latest point
+    # This is not a detection though
+    status, data = api(
+        'POST',
+        'photometry',
+        data={
+            'obj_id': obj_id,
+            'mjd': 99999.0,
+            'instrument_id': instrument_id,
+            'mag': None,
+            'magerr': None,
+            'limiting_mag': 22.3,
+            'magsys': 'ab',
+            'filter': 'ztfg',
+            'group_ids': [public_group.id],
+        },
+        token=upload_data_token,
+    )
+    assert status == 200
+    assert data['status'] == 'success'
+
+    # Another high mjd, but this time a photometry point not visible to the user
+    status, data = api(
+        'POST',
+        'photometry',
+        data={
+            'obj_id': obj_id,
+            'mjd': 99900.0,
+            'instrument_id': instrument_id,
+            'mag': None,
+            'magerr': None,
+            'limiting_mag': 22.3,
+            'magsys': 'ab',
+            'filter': 'ztfg',
+            'group_ids': [public_group2.id],
+        },
+        token=upload_data_token_two_groups,
+    )
+    assert status == 200
+    assert data['status'] == 'success'
+
+    # A high mjd, but lower than the first point
+    # Since this is a detection, it should be returned as "last_detected"
+    status, data = api(
+        'POST',
+        'photometry',
+        data={
+            'obj_id': obj_id,
+            'mjd': 90000.0,
+            'instrument_id': instrument_id,
+            'flux': 12.24,
+            'fluxerr': 0.031,
+            'zp': 25.0,
+            'magsys': 'ab',
+            'filter': 'ztfg',
+            'group_ids': [public_group.id],
+        },
+        token=upload_data_token,
+    )
+    assert status == 200
+    assert data['status'] == 'success'
+
+    status, data = api(
+        "GET",
+        "sources",
+        params={"includeDetectionStats": "true"},
+        token=view_only_token,
+    )
+    assert status == 200
+    assert data["status"] == "success"
+
+    assert any(
+        [
+            s["last_detected_at"] == arrow.get((90000.0 - 40_587) * 86400.0).isoformat()
+            for s in data["data"]["sources"]
+        ]
+    )
+    assert any(
+        [
+            s["peak_detected_at"] == arrow.get((90000.0 - 40_587) * 86400.0).isoformat()
```

Comment:
Should explain in a comment that `40_587` is the MJD of UNIX time 1970-01-01 and hence, since `arrow.get` views dates as seconds since UNIX time, s["peak_detected_at"]` is the MJD of 90000 in isodate format.

```
arrow.get("1970-01-01") - datetime.timedelta(40587)
# <Arrow [1858-11-17T00:00:00+00:00]>
```

Reply:
great idea!